sdk/core/allocator/main.cc

```diff
@@ -823,13 +830,18 @@ namespace
 	                                  void               *rawPointer,
 	                                  bool                reallyFree)
 	{
-		auto *capability = malloc_capability_unseal(heapCapability);
+		auto [capability, quotaPermissions] =
+		  malloc_capability_unseal(heapCapability);
 		if (capability == nullptr)
 		{
 			Debug::log<DebugLevel::Warning>("Invalid heap capability {}",
 			                                heapCapability);
 			return -EPERM;
 		}
```

Comment:
Is this check necessary?  I believe all of the code paths that fail to unseal in `malloc_capability_unseal` will return 0 for permissions so the following check will fail.

It might be cleaner for `malloc_capability_unseal` to take the required permissions and return either null or an unsealed capability that has the required permissions, rather than a pair.